platforms/chibios/drivers/wear_leveling/wear_leveling_efl.c

```diff
@@ -84,12 +84,14 @@ bool backing_store_init(void) {
 
     // Work out how many sectors we want to use, working backwards from the end of the flash
     flash_sector_t last_sector = desc->sectors_count - WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT;
+
+    // skip sectors that are past the actual flash size
+    while (flashGetSectorOffset(flash, last_sector) >= flash_size) {
+        last_sector--;
+    }
```

Comment:
Applying `WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT` first looks wrong (probably whoever added this code assumed that `desc->sectors_count` should be correct, but this might not be the case).  There are no real users of that option in the repo though.